File: 409-longest-palindrome/main.go
package main

func longestPalindrome(s string) int {

    n := len(s)
    freq := make(map[byte]int)

    for i := 0; i < n; i++{
        freq[s[i]] += 1
    }

    var length int = 0
    var oddFound bool = false

    for _, value := range freq {
        if value & 1 == 0{
            length += value
        }else{
            length += value - 1
            oddFound = true
        }
    }

    if oddFound {
        length += 1
    }

    return length
}

File: 409-longest-palindrome/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn longest_palindrome(s: String) -> i32 {
        
        let mut freq = HashMap::new();

        for ch in s.chars(){
            *freq.entry(ch).or_insert(0) += 1;
        }

    
        let mut length = 0;
        let mut odd_found = false;

        for &value in freq.values() {
            if value & 1 == 0 {
                length += value;
            }else {
                length += value - 1;
                odd_found = true;
            }
        }

        if odd_found {
            length += 1
        }

        length
    }
}
